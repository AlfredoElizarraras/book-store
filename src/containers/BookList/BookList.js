import React from 'react';
import PropTypes from 'prop-types';
import Book from '../../components/Book';
import './index.css';

const BookList = ({ books = [], removeBook }) => {
  const handleRemoveBook = id => {
    removeBook({ id });
  };

  return (
    <div className="bookList flex jc-c">
      <div style={{ borderBottom: '1px solid #e8e8e8', width: '80%' }}>
        {books.map(book => (
          <Book key={book.id} book={book} handleRemoveBook={handleRemoveBook} />
        ))}
      </div>
    </div>
  );
};

BookList.propTypes = {
  books: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      category: PropTypes.string.isRequired,
    }),
  ).isRequired,
  removeBook: PropTypes.func.isRequired,
};

export default BookList;
