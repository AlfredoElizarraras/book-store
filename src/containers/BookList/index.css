:root {
  --white: #fafafa;
  --black-5: rgba(0, 0, 0, 0.05);
}

.bookList {
  border-radius: 4px;
  border-top: solid 1px #e8e8e8;
  background-color: var(--white);
  padding: 2rem 0 2rem 2rem;
}
