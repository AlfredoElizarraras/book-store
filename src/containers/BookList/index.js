import { connect } from 'react-redux';
import BookList from './BookList';
import { removeBook, changeFilter } from '../../actions';
import { getBooksByCategory } from '../../utils/selectors/selectors';

const mapStateToProps = state => {
  const { filter } = state;
  const books = getBooksByCategory(state, filter);

  return { books };
};

export default connect(mapStateToProps, { removeBook, changeFilter })(BookList);
