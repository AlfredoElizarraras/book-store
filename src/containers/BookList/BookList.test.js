import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import BookList from './BookList';

describe('BookList', () => {
  const props = {
    books: [
      { id: 1, title: 'Book Title', category: 'Book Category' },
      { id: 2, title: 'Book Title One', category: 'Book Category One' },
    ],
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BookList books={props.books} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('Has a valid snapshot', () => {
    const component = renderer.create(<BookList books={props.books} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
