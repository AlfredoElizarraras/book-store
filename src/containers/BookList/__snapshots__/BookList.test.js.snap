// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList Has a valid snapshot 1`] = `
<table>
  <thead>
    <tr>
      <th>
        ID
      </th>
      <th>
        Title
      </th>
      <th>
        Category
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        1
      </td>
      <td>
        Book Title
      </td>
      <td>
        Book Category
      </td>
    </tr>
    <tr>
      <td>
        2
      </td>
      <td>
        Book Title One
      </td>
      <td>
        Book Category One
      </td>
    </tr>
  </tbody>
</table>
`;
